spec.md

```diff
@@ -3107,23 +3138,21 @@ provided `default` value instead of failing.
 
 ### hash
 
-`hash(x)` returns an integer hash of a string x
-such that two equal strings have the same hash.
+`hash(x)` returns an integer hash of a string or bytes x
+such that two equal values have the same hash.
 In other words `x == y` implies `hash(x) == hash(y)`.
+Any other type of argument in an error, even if it is suitable as the key of a dict.
+
 In the interests of reproducibility of Starlark program behavior over time and
-across implementations, the specific hash function is the same as that implemented by
+across implementations, the specific hash function for bytes is 32-bit FNV-1a,
```

Comment:
What's the merit of FNV-1a in particular? The merit of matching Java for strings is obvious.

Reply:
It's extremely simple and good enough.

Reply:
Sure, was just wondering if it was used in Go, or why -1a vs -1 . I'll trust your pick though.

Reply:
It's what I had already implemented. According to wikipedia, doing the XOR before the MUL leads to "slightly better avalanche characteristics".